frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/stateUpdate/vmSettings/vm-settings-tab-state-update.ts

```diff
@@ -103,6 +108,28 @@ export const flavorUpdater = ({ id, prevState, dispatch, getState }: UpdateOptio
   );
 };
 
+export const osUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
+  const state = getState();
+  if (!hasVMSettingsValueChanged(prevState, state, id, VMSettingsField.OPERATING_SYSTEM)) {
+    return;
+  }
+  const os = iGetVmSettingValue(state, id, VMSettingsField.OPERATING_SYSTEM);
+  const isWindows = os && os.startsWith('win');
+  const windowsTools = getStorages(state, id).find((storage) =>
+    Object.values(WINTOOLS_CONTAINER_NAMES).includes(getVolumeContainerImage(storage.volume)),
+  );
+  if (isWindows && !windowsTools) {
+    dispatch(
+      vmWizardInternalActions[InternalActionType.UpdateStorage](id, {
```

Comment:
there is no need for the spread as the object will be converted to immutable anyway